catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/provider/SolrProviderTestBase.java

```diff
@@ -0,0 +1,195 @@
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.source.solr.provider;
+
+import static org.junit.Assert.assertEquals;
+
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.AttributeDescriptorImpl;
+import ddf.catalog.data.impl.BasicTypes;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.SourceResponse;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.UnsupportedQueryException;
+import ddf.catalog.source.solr.SolrCatalogProvider;
+import ddf.catalog.source.solr.SolrProviderTest;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.commons.lang.StringUtils;
+import org.joda.time.DateTime;
+import org.opengis.filter.Filter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class SolrProviderTestBase {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SolrProviderQuery.class);
+
+  static final int ONE_HIT = 1;
+
+  static final int ALL_RESULTS = -1;
+
+  static final String DEFAULT_TEST_ESCAPE = "\\";
+
+  static final String DEFAULT_TEST_SINGLE_WILDCARD = "?";
+
+  static final String DEFAULT_TEST_WILDCARD = "*";
+
+  protected static SolrCatalogProvider provider = SolrProviderTest.getProvider();
+
+  protected static SolrFilterBuilder filterBuilder = new SolrFilterBuilder();
+
+  private static final int TEST_METHOD_NAME_INDEX = 3;
+
+  private static final int ONE_SECOND = 1;
+
+  private static final int TIME_STEP_10SECONDS = 10 * ONE_SECOND;
+
+  private static final int A_LITTLE_WHILE = TIME_STEP_10SECONDS;
+
+  protected static void deleteAll() throws IngestException, UnsupportedQueryException {
+    deleteAll(TEST_METHOD_NAME_INDEX);
+  }
+
+  protected static void deleteAll(int methodNameIndex)
+      throws IngestException, UnsupportedQueryException {
+    messageBreak(Thread.currentThread().getStackTrace()[methodNameIndex].getMethodName() + "()");
```

Comment:
❓ Could we use JUnit's `TestName` rule instead of parsing the stack trace and relying on a potentially wrong index?